### PR TITLE
Ana/all new proposals components API changes

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Splitting #4566 . API part

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: api/lib/reducers/polkadotV0-reducers.js [hunk 1]
```diff
@@ -519,12 +519,25 @@ function networkAccountReducer(account) {
...
       account && account.identity && account.identity.display
         ? account.identity.display
-        : account.accountId,
+        : account.accountId || '',
     address: account && account.accountId ? account.accountId : '',
     picture: account ? account.twitter : '' // TODO: get the twitter picture using scriptRunner
   }
 }
 
+function getProposalSummary(type) {
```
Comment (faboweb) [inline api/lib/reducers/polkadotV0-reducers.js:528]: why are those duplicated?

File: api/lib/source/polkadotV0-source.js [hunk 2]
```diff
@@ -1214,10 +1216,7 @@ class polkadotAPI {
       ),
       topVoters: await Promise.all(
         topVoters.map(async (topVoterAddress) => {
-          const accountInfo = await this.getNetworkAccountInfo(
-            topVoterAddress,
-            api
-          )
+          const accountInfo = await api.derive.accounts.info(topVoterAddress)
```
Comment (faboweb) [inline api/lib/source/polkadotV0-source.js:1219]: didn't we discuss, that here we can also use `getNetworkAccountInfo`?

Comment (Bitcoinera) [reply api/lib/source/polkadotV0-source.js]: true, I don't know what happened here. Probably was changed in develop but the original PR this one came from was older. Thank you for remembering this

File: api/lib/reducers/polkadotV0-reducers.js (hunk 1 above)
Comment (Bitcoinera) [reply api/lib/reducers/polkadotV0-reducers.js]: which ones?

File: api/lib/reducers/common.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+      case `TEXT`:
+        return `This is a text proposal. Text proposals can be proposed by anyone and are used as a signalling mechanism for this community. If this proposal is accepted, nothing will change without community coordination.`
+      case `PARAMETER_CHANGE`:
+        return `This is a parameter change proposal. Parameter change proposals can be proposed by anyone and include changes to the code of this network. If this proposal is approved the underlying code will change.`
+      case `TREASURY`:
+        return `This is a treasury proposal. Treasury proposals can be proposed by anyone and are a request for funds from the treasury / community pool.`
+      default:
+        return `Unknown proposal type`
+    }
+}
```
Comment (Bitcoinera) [inline api/lib/reducers/common.js:12]: Ah nice, much better like this, that's right!

File: api/lib/reducers/polkadotV0-reducers.js (hunk 1 above)
Comment (Bitcoinera) [reply api/lib/reducers/polkadotV0-reducers.js]: ah, you meant the `getProposalSummary` function. Yeah, better like you did, definitely

**Pull request merged**